src/main/java/NyanBot.java

```diff
@@ -0,0 +1,237 @@
+        }
+    }
+
+    private static void markTask(String input) {
+        try {
+            String[] splitInputs = input.split(" ");
+            int index = Integer.parseInt(splitInputs[1]) - 1;
+            tasks.get(index).markAsDone();
+            Printer.printMarkSuccess();
+        } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
+            Printer.printNullError();
+        } catch (NumberFormatException e) {
+            Printer.printMarkUsage();
+        }
+    }
+
+    private static void unmarkTask(String input) {
+        try {
+            String[] splitInputs = input.split(" ");
+            int index = Integer.parseInt(splitInputs[1]) - 1;
+            tasks.get(index).markAsUndone();
+            Printer.printUnmarkSuccess();
+        } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
+            Printer.printNullError();
+        } catch (NumberFormatException e) {
+            Printer.printUnmarkUsage();
+        }
+    }
+
+    private static void addTodo(String input) {
+        try {
+            String description = input.substring(5);
+            Todo todo = new Todo(description);
+            addTask(todo);
+        } catch (StringIndexOutOfBoundsException e) {
+            Printer.printMissingDescription();
+            Printer.printTodoUsage();
+        }
+    }
+
+    private static void addDeadline(String input) {
+        try {
+            String[] splitInputs = input.split("/");
+            String description = splitInputs[0].substring(9);
+            String date = splitInputs[1];
+            Deadline deadline = new Deadline(description, date);
+            addTask(deadline);
+        } catch (StringIndexOutOfBoundsException e) {
+            Printer.printMissingDescription();
+            Printer.printDeadlineUsage();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            Printer.printMissingDate();
+            Printer.printDeadlineUsage();
+        }
+    }
+
+    private static void addEvent(String input) {
+        try {
+            String[] splitInputs = input.split("/");
+            String description = splitInputs[0].substring(6);
+            String start = splitInputs[1];
+            String end = splitInputs[2];
+            Event event = new Event(description, start, end);
+            addTask(event);
+        } catch (StringIndexOutOfBoundsException e) {
+            Printer.printMissingDescription();
+            Printer.printEventUsage();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            Printer.printMissingStartEnd();
+            Printer.printEventUsage();
+        }
+    }
+
+    private static String parseCommand(String input) {
+        String[] splitInputs = input.split(" ");
+        return splitInputs[0].toUpperCase();
+    }
+
+    private static void startNyan() {
+        readFile();
+        Printer.printGreet();
+    }
+    private static void getInput() {
+        Scanner scanner = new Scanner(System.in);
```

Comment:
this function is a little longer than 30 lines. see if you can abstract something out into another function 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods

Reply:
Hi Vishruti thanks for the time taken to read my code

Reply:
Hi Vishruti thanks for the time taken to review my code, managed to implement all your improvements except this one

The getInput() method has too many lines due to the switch cases. I managed to separate the logic and scanner methods to refactor into the following processCommand() method:

`private static void processCommand(String input) {
        String command = parseCommand(input);
        switch (command) {
            case BYE_COMMAND:
                isRunning = false;
                break;
            case DELETE_COMMAND:
                deleteTask(input);
                break;
            case LIST_COMMAND:
                Printer.printTasks(tasks);
                break;
            case MARK_COMMAND:
                markTask(input);
                break;
            case UNMARK_COMMAND:
                unmarkTask(input);
                break;
            case TODO_COMMAND:
                addTodo(input);
                break;
            case DEADLINE_COMMAND:
                addDeadline(input);
                break;
            case EVENT_COMMAND:
                addEvent(input);
                break;
            case HELP_COMMAND:
                Printer.printSike();
                break;
            default:
                Printer.printInvalidInput();
                break;
        }
    }`
    
    I think it is still >30 lines so do you have any recommendations for this case

Reply:
Looks like it is around 34 lines right?
![image](https://github.com/nus-cs2113-AY2324S2/ip/assets/57522870/1352f742-0327-43f6-ae67-ecbcc710da50)

It should be fine, it is quite readable and only goes 4 lines above the max